src/coro/coro.ml

```diff
@@ -315,12 +318,12 @@ let fun_to_coro ctx coro_type =
 	]) basic.tvoid null_pos in
 
 	let tf_args = args @ [ (vcompletion,None) ] in
-	let tf_type = basic.tany in
+	let tf_type = basic.tany in (* TODO: this should be basic.tcoro.continuation_result, but cpp hates it *)
```

Comment:
I'm having a look at this now, issue seems to be that the continuation result class is not forward declared in the header but it is included in the implementation file. Might be another case of the forward declaration logic needing an update.

Reply:
The forward reference gathering logic wasn't expanding the coroutine type so it never "saw" the continuation result type since it's hidden on the unexpanded coroutine type.